nistats/second_level_model.py

```diff
@@ -262,6 +269,8 @@ def fit(self, second_level_input, confounds=None, design_matrix=None):
             sample_map = second_level_input['effects_map_path'][0]
             labels = second_level_input['subject_label']
             subjects_label = labels.values.tolist()
+        elif isinstance(second_level_input, Nifti1Image):
+            sample_map = mean_img(second_level_input)
```

Comment:
Wasn't it the idea of operating on list of 3D images given a 4D image ? But, it seems you are averaging.

Reply:
The idea was to avoid using second_level_input[0], as second_level_input is not necessarily a list. Also, IIRC the sample_map is used for masking, hence it is actually better to base this on the mean image. Does that make sense ?